build/CHANGELOG.md

```diff
@@ -1,3 +1,12 @@
+## 1.2.0
+
+- Add `BuildStep.removeDependency(AssetId id)` method to `BuildStep` which
+  allows you to delete a dependency on some input after the fact.
+  - This should be used very carefully as it can easily lead to correctness
```

Comment:
You probably want to always rerun on existence changes; in bazel these change the actual command line of the action, guaranteeing a rerun. The action can do something with that data before it decides to read anything, e.g. figuring out which files to read based on what exists.

Reply:
Unfortunately right now we don't have a way of tracking existence of a file explicitly - it is either an input or not and we incorporate existence into the content hash.

It is true I suppose though that a kernel action would fail if some transitive dep no longer existed even if it wasn't used?

Reply:
I added one mitigation to the actual kernel/ddk worker implementations to mitigate the risk by not reporting unused inputs if the action failed based on offline conversations.

The issue around asset existence is real but I don't believe it will actually cause any meaningful problems - it can cause non-hermetic _failed_ builds but not non-hermetic _successful_ builds, if that makes sense.